apps/src/code-studio/components/header/ProjectRemix.jsx

```diff
@@ -0,0 +1,62 @@
+/* globals dashboard, appOptions */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import {refreshProjectName} from '../../headerRedux';
+
+class ProjectRemix extends React.Component {
+  static propTypes = {
+    isSignedIn: PropTypes.bool,
+    lightStyle: PropTypes.bool,
+    refreshProjectName: PropTypes.func.isRequired
+  };
+
+  remixProject = () => {
+    if (
+      dashboard.project.getCurrentId() &&
+      dashboard.project.canServerSideRemix()
+    ) {
+      dashboard.project.serverSideRemix();
+    } else if (!this.props.isSignedIn) {
+      window.location.assign(
+        `/users/sign_in?user_return_to=${window.location.pathname}`
+      );
+    } else {
+      // We don't have an id. This implies we are either on a legacy /c/ share
+      // page or a script level. In these cases, copy will create a new project
+      // for us.
+      const newName =
+        'Remix: ' +
+        (dashboard.project.getCurrentName() ||
+          appOptions.level.projectTemplateLevelName ||
+          'My Project');
+      dashboard.project
+        .copy(newName, {shouldNavigate: true})
+        .then(() => this.props.refreshProjectName())
+        .catch(err => console.log(err));
+    }
+  };
+
+  render() {
+    let className = 'project_remix header_button';
+    // TODO do this with inline styles
```

Comment:
New TODO.

Reply:
+1, I believe TODOs should ideally be removed, but if not then should at least have your name on them.

Reply:
I'll replace these TODOs with some jira cards